### PR TITLE
Fixes 4475,4438: routing issues when deleting popular repos

### PR DESCRIPTION
**Pull request by xbhouse**
## Summary

This fixes the routing issues when trying to removing popular repos by adding a delete route specific to popular repos. 

## Testing steps

- When clicking Remove on one popular repository, this should route to the delete modal and show the selected repo. This is currently broken in stage preview and prod preview and returns a 404. Make sure to test this in preview environments (for devs, npm run local > stage > beta or npm run local > prod > beta)
- When trying to delete popular repos by bulk selecting, this should route to the delete modal and show the selected repos
- If a repo has associated templates, clicking on the template should route to the custom repos step in the template modal
- Also fixes the positioning of the expanded content (list of templates) in the delete modal if a repo is associated with more than 3 templates

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4475

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4438

File: src/Pages/Repositories/ContentListTable/components/DeleteContentModal/DeleteContentModal.tsx [hunk 1]
```diff
@@ -250,6 +259,20 @@ export default function DeleteContentModal() {
                                 </ListItem>
                               ))}
                           </ExpandableSection>
+                          <ExpandableSectionToggle
+                            isExpanded={expandState[index]}
+                            onToggle={() =>
+                                setExpandState((prev) => ({ ...prev, [index]: !prev[index] }))
+                                }
+                            toggleId='detached-expandable-section-toggle'
+                            contentId='detached-expandable-section-content'
```
Comment (Andrewgdewar) [inline src/Pages/Repositories/ContentListTable/components/DeleteContentModal/DeleteContentModal.tsx:268]: This toggleId and contentId is duplicated. 
You may want an "_up"/"_down" on the ids here.

File: src/Pages/Repositories/PopularRepositoriesTable/PopularRepositoriesTable.tsx [hunk 2]
```diff
@@ -556,7 +562,7 @@ const PopularRepositoriesTable = () => {
             <Thead>
               <Tr>
                 <Hide hide={!rbac?.repoWrite}>
-                  <Th
+                  <Th screenReaderText='empty'
```
Comment (Andrewgdewar) [inline src/Pages/Repositories/PopularRepositoriesTable/PopularRepositoriesTable.tsx:565]: although this works a better method to remove the console error is to add an aria-label for the screen reader to read as seen [here](https://github.com/xbhouse/content-sources-frontend/blob/758d1f0aa4abd51573068c35d24343f5222fae66/src/Pages/Repositories/ContentListTable/ContentListTable.tsx#L534-L535)

File: src/Routes/Repositories/useRepositoryRoutes.tsx [hunk 3]
```diff
@@ -53,6 +53,13 @@ export default function useRepositoryRoutes(): TabbedRouteItem[] {
         title: 'Popular repositories',
         route: POPULAR_REPOSITORIES_ROUTE,
         Element: PopularRepositoriesTable,
+        ChildRoutes: [
```
Comment (Andrewgdewar) [inline src/Routes/Repositories/useRepositoryRoutes.tsx:56]: in this case on could remove the outer layer here.

hasWrite  ? [
                  { path: DELETE_ROUTE, Element: DeleteContentModal },
                ]
              : []

**Review by Andrewgdewar — APPROVED** [review]
A few minor #nits, otherwise ACK!

File: src/Pages/Repositories/ContentListTable/components/DeleteContentModal/DeleteContentModal.tsx (hunk 1 above)
Comment (xbhouse) [reply src/Pages/Repositories/ContentListTable/components/DeleteContentModal/DeleteContentModal.tsx]: pushed up fixes for this and the comments below

**Pull request merged**